src/main/java/pl/javastart/task/app/UniversityApp.java

```diff
@@ -0,0 +1,313 @@
+            }
+        } else {
+            System.out.println("Prowadzący o id " + lecturerId + " nie istnieje");
+        }
+    }
+
+    private void addGroup(Group group) {
+        if (countGroups <= groups.length) {
+            groups[countGroups] = group;
+            countGroups++;
+        } else {
+            groups = Arrays.copyOf(groups, groups.length * 2);
+        }
+    }
+
+    private boolean checkGroupExists(String code) {
+        Group groupByCode = findGroupByCode(code);
+        return groupByCode != null;
+    }
+
+    private Group findGroupByCode(String code) {
+        Group foundGroup = null;
+        for (int i = 0; i < countGroups; i++) {
+            Group group = groups[i];
+            if (group.getCode().equals(code)) {
+                foundGroup = group;
+            }
+        }
+        return foundGroup;
+    }
+
+    /**
+     * Dodaje studenta do grupy zajęciowej.
+     * W przypadku gdy grupa zajęciowa nie istnieje wyświetlany jest komunikat:
+     * "Grupa [kod grupy] nie istnieje
+     *
+     * @param index     - unikalny numer indeksu studenta
+     * @param groupCode - kod grupy utworzonej wcześniej za pomocą {@link #createGroup(String, String, int)}
+     * @param firstName - imię studenta
+     * @param lastName  - nazwisko studenta
+     */
+    public void addStudentToGroup(int index, String groupCode, String firstName, String lastName) {
+        Group foundGroup = findGroupByCode(groupCode);
+        if (foundGroup != null) {
+            if (foundGroup.containsStudent(index)) {
+                System.out.println("Student o indeksie " + index + " jest już w grupie " + groupCode);
+            } else {
+                Student student = new Student(index, firstName, lastName);
+                addUniqueStudent(student); //dodaje studenta do tablicy unikalnych studentów
+                foundGroup.addStudent(student); //dodaje do konkretnej grupy
+            }
+        } else {
+            System.out.println("Grupa " + groupCode + " nie istnieje");
+        }
+    }
+
+    private void addUniqueStudent(Student student) {
+        boolean found = findUniqueStudent(student);
+        if (!found) {
+            if (countAllStudents <= allStudents.length) {
+                allStudents[countAllStudents] = student;
+                countAllStudents++;
+            } else {
+                allStudents = Arrays.copyOf(allStudents, allStudents.length * 2);
+            }
+        }
+    }
+
+    private boolean findUniqueStudent(Student student) {
+        boolean found = false;
+        for (int i = 0; i < countAllStudents; i++) {
+            if (allStudents[i].getIndex() == student.getIndex()) {
+                showMessage(allStudents[i].getIndex());
+                found = true;
+            }
+        }
+        return found;
+    }
+
+    private static void showMessage(int index) {
+        System.out.println("Student o indeksie " + index + " już istnieje");
+    }
```

Comment:
Jeżeli metoda opakowuje zwykłego System.out.pintlna, to raczej bym ją usunął i zostawił sout po prostu w metodzie findUniqueStudent